src/protocol/requests/fetch/v4/response.js

```diff
@@ -7,6 +7,7 @@ const { MAGIC_BYTE } = require('../../../recordBatch/v0')
 // the magic offset is at the same offset for all current message formats, but the 4 bytes
 // between the size and the magic is dependent on the version.
 const MAGIC_OFFSET = 16
+const RECORD_BATCH_OVERHEAD = 49
```

Comment:
Consider if maybe the decoders should expose how many bytes they require, since we have these checks wherever you're decoding a list of something.